deploy-manage/users-roles/cluster-or-deployment-auth/file-based.md

```diff
@@ -14,107 +14,156 @@ products:
+You don’t need to explicitly configure the `file` realm. The `file` and `native` realms are added to the realm chain by default. Unless configured otherwise, the `file` realm is added first, followed by the `native` realm. You can define only one `file` realm per node. To override the default `file` realm configuration
 
 1. (Optional) Add a realm configuration to [`elasticsearch.yml`](/deploy-manage/stack-settings.md) under the `xpack.security.authc.realms.file` namespace. At a minimum, you must set the realm’s `order` attribute.
 
     For example, the following snippet shows a `file` realm configuration that sets the `order` to zero so the realm is checked first:
 
     ```yaml
-    xpack:
-      security:
-        authc:
-          realms:
-            file:
-              file1:
-                order: 0
+    xpack.security.authc.realms.file.file1.order: 0
     ```
 
-2. If you're using a self-managed {{es}} cluster, optionally change how often the `users` and `users_roles` files are checked.
+2. (Optional) For self-managed deployments, you may change how often `file` realm backing files are checked.
 
-    By default, {{es}} checks these files for changes every 5 seconds. You can change this default behavior by changing the `resource.reload.interval.high` setting in the [`elasticsearch.yml`](/deploy-manage/stack-settings.md) file.
+    By default, {{es}} checks these files for changes every 5 seconds. You can override this default behavior by changing the `resource.reload.interval.high` setting's value in the [`elasticsearch.yml`](/deploy-manage/stack-settings.md) file.
 
     :::{{warning}}
-    Because `resource.reload.interval.high` is a common setting in {{es}}, changing its value may effect other schedules in the system.
+    Because `resource.reload.interval.high` is a foundational setting for {{es}}, changing its value may effect other schedules in the system.
     :::
 
-3. Restart {{es}}.
+3. In self-managed deployments, if either prior setting is modified, you will need to [rolling restart](/deploy-manage/maintenance/start-stop-services/full-cluster-restart-rolling-restart-procedures#restart-cluster-rolling) those {{es}} nodes for your changes to take effect. In {{eck}}, changes are automatically propagated.
 
-   In {{eck}}, this change is propagated automatically.
+## Configure files [file-realm-files]
 
+The `file` realm reads its files upon the local node's initial startup and as periodically refreshed based on the `resource.reload.interval.high` setting. You do not need to restart nodes for changes to take effect. Its files are located under the [`ES_PATH_CONF` directory](/deploy-manage/deploy/self-managed/configure-elasticsearch#config-files-location) and contain
 
-## Add users
+* `roles.yml` for [defining roles](/deploy-manage/users-roles/cluster-or-deployment-auth/defining-roles)
```

Comment:
Role configuration is not part of the file realm. IIUC it is separate mechanism and roles defined through it can be referenced in users create via the native realm as well.